runtime/vm/vmthread.cpp

```diff
@@ -1964,6 +1964,16 @@ startJavaThreadInternal(J9VMThread * currentThread, UDATA privateFlags, UDATA os
 	omrthread_set_name(newThread->osThread, threadName);
 #endif
 
+#if defined(J9VM_OPT_CRIU_SUPPORT)
+	if (VM_CRIUHelpers::isJVMInSingleThreadMode(vm) && VM_VMHelpers::threadCanRunJavaCode(newThread)) {
+		/* If JDWP is enabled, new JDWP debug threads should not be halted. */
+		if (vm->checkpointState.isJdwpEnabled && (0 == strncmp("JDWP", threadName, 4))) {
+			clearHaltFlag(newThread, J9_PUBLIC_FLAGS_HALT_THREAD_FOR_CHECKPOINT);
```

Comment:
Can `allocateVMThread()` be modified to skip `setHaltFlag()` instead?
https://github.com/eclipse-openj9/openj9/blob/d4443d666d69af7475669d5f1537da1f1234dda5/runtime/vm/vmthread.cpp#L295

Reply:
Added new parameter to `allocateVMThread` in https://github.com/eclipse-openj9/openj9/commit/508da49ef7c69fe039d046f19b4e2d1a1bffa4c2.

Now the code does a check if thread name starts with `JDWP` before setting halt flag.